frontend/src/utils/colors.ts

```diff
@@ -0,0 +1,93 @@
+// The main idea behind defining these color palettes is to abstract away
+// from knowing individual material design colors. Right now, the system
+// is a bit lacking, but hopefully this can also be expanded to handle
+// light/night mode etc.
+
+// => Components implement less conditional styles themselves for the basic stuff
+
+/** Color palette defining multiple colors based on Material design */
+export type Palette = 'primary' | 'secondary' | 'tertiary' | 'red';
+
+/** Background color for this palette */
+export const bgColor = (c: Palette) => {
+    switch (c) {
+        case 'primary':
+            return 'bg-primary';
+        case 'secondary':
+            return 'bg-secondary';
+        case 'tertiary':
+            return 'bg-tertiary';
+        case 'red':
+            return 'bg-red';
+    }
+};
+
+/** Container color for this palette */
+export const containerColor = (c: Palette) => {
+    switch (c) {
+        case 'primary':
+            return 'bg-primary-90';
+        case 'secondary':
+            return 'bg-secondary-90';
+        case 'tertiary':
+            return 'bg-tertiary-90';
+        case 'red':
+            return 'bg-red-90';
+    }
+};
+
+/** Hover color for the container of this palette */
+export const hoverColor = (c: Palette) => {
```

Comment:
Just specifying 'color' in these is quite ambiguous, does it refer to text color or bg color? Border color? Some other color?

Reply:
Renamed color methods:
`bgColor => bg`
`containerColor => bgContainer`
`hoverColor => bgHover`
`activeColor => bgActive`
`onColor => textOnBg`
`onContainer => textOnContainer`